frontend/src/app/cities-upload/cities-upload.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component } from '@angular/core';
+import { SearchCityService } from '../search-city.service';
+import { HttpEventType } from '@angular/common/http';
+import { Observable } from "rxjs/Observable";
+import 'rxjs/add/observable/interval';
+import 'rxjs/add/operator/finally';
+import 'rxjs/add/operator/take';
+import 'rxjs/add/operator/takeWhile';
+
+const ESTIMATED_PROCESSING_TIME_IN_MILLIS = 20000;
```

Comment:
Sure, it's not necessary here, I was just wondering what would be the best way. It's a bit fuzzy in my head, but maybe something like project reactor/spring 5 could help to do this.